File: WebApplication/ClientApp/src/app/models/city.ts
import { IBase } from "./base";
import { IEmpire } from "./empire";
import { IRegion } from "./region";

export interface ICity extends IBase{
    name: string,
    population: number,
    empire: IEmpire,
    empireId: number,
    region: IRegion,
    regionId: number
}

File: Core/Entities/City.cs
using System.Text.Json.Serialization;

namespace Core.Entities
{
    public class City:EntityBase
    {
        public string Name { get; set; }
        public int Population { get; set; }
        [JsonIgnore]
        public Empire Empire { get; set; }
        public long EmpireId { get; set; }
        public Region Region { get; set; }
        public long RegionId { get; set; }
    }
}